### ISSUE TITLE
shortcut for Iterable<E: Comparable<E>>.min()

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#163]
*Platform* (jvm, jdk8, js, android): all

## Code related feature
```
expect(listOf(1,2,3)).min().toBe(3) 
expect(listOf(1,2,3)).min {
   isGreaterThan(3)  
   isLessThan(5)
}

//instead of

expect(listOf(1,2,3)).feature { f(it::min) }.notToBeNull().toBe(3)
expect(listOf(1,2,3)).feature({ f(it::min) }) {
  notToBeNull {
    isGreaterThan(3)  
    isLessThan(5)
  }
}
```
However, in reporting we want to see the following in case there aren't any elements in the iterable:
```
expect: List()
* min: !! cannot be determined, empty Iterable
  >> to be: 3
```

Following the things you need to do:

*domain*
- extend IterableAssertions with a function `min` (see ListAssertions.get as a guideline)
- modify IterableAssertionsBuilder, delegate to iterableAssertions (see other functions as a guideline)
- delegate implementation to robstoll-lib in IterableAssertionsImpl (see other functions as a guideline)

*lib*
- implement _min in iterableAssertions by using `ExpectImpl.feature.extractor` (see listAssertions -> _get) => use the following as failure representation
  - add `NO_ELEMENTS("❗❗ cannot be determined, empty Iterable")` to DescriptionIterable of en_GB
  - add `NO_ELEMENTS("❗❗ kann nicht eruiert werden, leeres Iterable")` to DescriptionIterable of de_CH

*api*  (only fluent-en_GB)
- provide a val which returns Expect<E> (see collectionAssertions.kt as a guideline)
- provide a fun which expects an `assertionCreator`-lambda and returns Expect<T> where `T : Iterable<E>, E : Comparable<E>` (see collectionAssertions.kt as a guideline)
- augment IterableFeatureAssertionsSpec with `min` in specs-common (see for instance CollectionFeatureAssertionsSpec) and extend it in atrium-api-fluent-en_GB-common/src/test

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime,
  we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

**Comment by ntedgi** [comment]
I'll work on this

**Comment by piyushmor** [comment]
is someone still working on this? else I will start

**Comment by robstoll** [comment]
@ntedgi please answer if you are. @piyushmor feel free to start, in the worst case you both do the same work

**Comment by ntedgi** [comment]
sorry didn't manged to finish that yet

**Comment by robstoll** [comment]
@ntedgi did you already start? If so, you could still push your changes as WIP PR and @piyushmor could take over from there. Up to you two

**Comment by piyushmor** [comment]
Would totally like to take over

**Comment by piyushmor** [comment]
@robstoll  Kindly check the PR again. If the changes are fine. I have issues in adding the Feature Test

**Comment by robstoll** [comment]
What issues? I'll take a look now

**Comment by piyushmor** [comment]
I took `size` as a reference, which returns an `Expect<Iterable<E>>` whereas get returns an `Expect<E>`. Not sure why the test expects a different behavior. will commit the test even if compilation breaks once you confirm what i have done so far is fine.

**Comment by robstoll** [comment]
Please push the code as well, hard to tell what you mean without further context

**Comment by robstoll** [comment]
Implemented by da897cb2322872452deb0e22ef875c16043a3b9d
